cmd/events/events.go

```diff
@@ -0,0 +1,161 @@
+	selectorFlags.BoolVarP(&selectorOpts.follow, "follow", "f", false, "Follow agent events output")
+	eventsCmd.PersistentFlags().AddFlagSet(selectorFlags)
+
+	// formatting flags
+	formattingFlags := pflag.NewFlagSet("Formatting", pflag.ContinueOnError)
+	formattingFlags.StringVarP(
+		&formattingOpts.output, "output", "o", "",
+		`Specify the output format, one of:
+  compact:  Compact output
+  dict:     Each agent event is shown as KEY:VALUE pair
+  json:     JSON encoding
+  jsonpb:   Output each GetAgentEventsResponse according to proto3's JSON mapping
+  table:    Tab-aligned columns`)
+	formattingFlags.BoolVarP(&formattingOpts.nodeName, "print-node-name", "", false, "Print node name in output")
+	formattingFlags.StringVar(
+		&formattingOpts.timeFormat, "time-format", "StampMilli",
+		fmt.Sprintf(`Specify the time format for printing. This option does not apply to the json and jsonpb output type. One of:
+  StampMilli:   %s
+  RFC3339:      %s
+  RFC3339Milli: %s
+  RFC3339Micro: %s
+  RFC3339Nano:  %s
+  RFC1123Z:     %s
+ `, time.StampMilli, time.RFC3339, hubtime.RFC3339Milli, hubtime.RFC3339Micro, time.RFC3339Nano, time.RFC1123Z),
+	)
+	eventsCmd.PersistentFlags().AddFlagSet(formattingFlags)
+
+	// other flags
+	otherFlags := pflag.NewFlagSet("other", pflag.ContinueOnError)
+	otherFlags.BoolVarP(&otherOpts.ignoreStderr, "silent-errors", "s", false, "Silently ignores errors and warnings")
+	eventsCmd.PersistentFlags().AddFlagSet(otherFlags)
+
+	// advanced completion for flags
+	eventsCmd.RegisterFlagCompletionFunc("time-format", func(_ *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
+		return hubtime.FormatNames, cobra.ShellCompDirectiveDefault
+	})
+
+	// add config.ServerFlags to the help template as these flags are used by
+	// this command
+	eventsCmd.SetUsageTemplate(template.Usage(selectorFlags, formattingFlags, config.ServerFlags, otherFlags))
+
+	eventsCmd.AddCommand(
+		newAgentEventsCommand(vp),
+	)
+	return eventsCmd
+}
+
+func handleArgs(debug bool) error {
+	// initialize the printer with any options that were passed in
+	var opts = []hubprinter.Option{
+		hubprinter.WithTimeFormat(hubtime.TimeFormatNameToLayout(formattingOpts.timeFormat)),
+	}
+
+	switch formattingOpts.output {
+	case "compact":
+		opts = append(opts, hubprinter.Compact())
+	case "dict":
+		opts = append(opts, hubprinter.Dict())
+	case "json", "JSON":
+		opts = append(opts, hubprinter.JSON())
+	case "jsonpb":
+		opts = append(opts, hubprinter.JSONPB())
+	case "tab", "table":
+		if selectorOpts.follow {
+			return fmt.Errorf("table output format is not compatible with follow mode")
+		}
+		opts = append(opts, hubprinter.Tab())
+	case "":
+		// no format specified, choose most appropriate format based on
+		// user provided flags
+		if selectorOpts.follow {
+			opts = append(opts, hubprinter.Compact())
+		} else {
+			opts = append(opts, hubprinter.Tab())
+		}
```

Comment:
no change required, but if https://github.com/cilium/hubble/pull/536 gets merged before this, we'll likely want to default to compact here as well

Reply:
Given that #536 is already approved and likely to be merged, I've changed the default to compact here as well.